src/components/Breadcrumbs/Breadcrumbs.tsx

```diff
@@ -1,92 +1,101 @@
-  blockName?: string;
   /** Breadcrumb links as an array */
-  breadcrumbs: breadcrumb[] | JSX.Element[];
+  breadcrumbsData: BreadcrumbstDataProps[];
   /** className you can add in addition to 'input' */
   className?: string;
-  /** Modifiers array for use with BEM. See how to work with modifiers and BEM here: http://getbem.com/introduction/ */
-  modifiers?: string[];
+  /** ID that other components can cross reference for accessibility purposes */
+  id?: string;
+  /** Used to control how the `Hero` component will be rendered. */
+  colorVariant?: ColorVariants;
 }
 
-export default class Breadcrumbs extends React.Component<BreadcrumbProps, any> {
-  static defaultProps = {};
+const getElementsFromData = (data, breadcrumbsID) => {
+  const breadcrumbItems = [];
 
-  constructor(props: BreadcrumbProps) {
-    super(props);
+  if (!data?.length) {
+    return {};
   }
 
-  render(): JSX.Element {
-    const { breadcrumbs, className, blockName, modifiers } = this.props;
-
-    const baseClass = "breadcrumbs";
-
-    const breadcrumbItems = [];
-
-    if (!breadcrumbs || breadcrumbs.length === 0) {
-      throw new Error(
-        "Breadcrumbs must contain a set of links. Breadcrumbs currently empty"
-      );
-    }
-    breadcrumbs.forEach((item: breadcrumb | JSX.Element, index: number) => {
-      const last = index === breadcrumbs.length - 1;
-      let linkComponent: string | JSX.Element;
-
-      if (isTextBreadcrumb(item)) {
-        linkComponent = item.url ? (
-          <a href={item.url} className={bem("link", [], baseClass)}>
-            {item.text}
-          </a>
-        ) : (
-          item.text
-        );
-      } else {
-        const props = {
-          className: bem("link", [], baseClass),
-        };
-        linkComponent = React.createElement(
-          item.type,
-          { ...props, ...item.props },
-          item.props.children
-        );
-      }
-      breadcrumbItems.push(
-        <li
-          key={`${baseClass}-${index}`}
-          className={bem("item", [], baseClass)}
-        >
-          {last && (
+  data.map((breadcrumbData, index) => {
+    const tempCrumb = (
+      <BreadcrumbItem
+        key={index}
+        isCurrentPage={index === data.length - 1 ? true : false}
+      >
+        <BreadcrumbLink href={breadcrumbData.url}>
+          {index === data.length - 2 && (
             <Icon
               name={IconNames.arrow}
-              blockName={baseClass}
-              modifiers={["small"]}
+              size={IconSizes.small}
+              iconRotation={IconRotationTypes.rotate90}
+              id={`${breadcrumbsID}__backarrow`}
               decorative={true}
+              titleText={breadcrumbData.text}
             />
           )}
-          {linkComponent}
-        </li>
-      );
-    });
+          <span className="breadcrumb-label">{breadcrumbData.text}</span>
```

Comment:
I see this was added to target it in the CSS JS object. Eventually, it'd be good to find a non-classname way of doing it.

Reply:
Right.  I needed a way to separate the arrow icon from the text label.  If you have any ideas about js related methods, let me know.